### PR TITLE
__Create Budget App__

### PR DESCRIPTION
**Pull request by martinkarugaba**
# Changes made

## Design
- [ ] Followed design guidelines, including:
Colors.
Typography: font face, size and weight.
Layout: composition and space between elements.

## Splash screen

- [ ] Added simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
Sign up and log in pages

- [ ] The user can register in the app with full name, email and password (all mandatory).
- [ ]The user can log into the app using email and password.
- [ ]If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

## Home page (categories page)

- [ ] When the user logs in, they are presented with the categories page.
- [ ] For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
- [ ] When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
- [ ] There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

## Transactions page

- [ ] For a given category, the list of transactions is presented, ordered by the most recent.
- [ ] At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
- [ ] There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
- [ ] When the user clicks on the "Back" button (<), the user navigates to the home page.
"Add a new category" page

- [ ] The user fills out a form to create a new category, indicating their name and icon (both mandatory).
- [ ] The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
- [ ] When the user clicks on the "Back" button (<), the user navigates to the home page.
"Add a new transaction" page

- [ ] The user fills out a form to create a new transaction with:
name (mandatory)
amount (mandatory)
categories (mandatory at least one)
- [ ] The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
- [ ] When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
 
## Testing requirements
- [ ] Created unit and integration tests for all the most important components of your RoR application
- [ ] Added README and LICENSE file
- [ ] Deployed the project to [render.com](https://mysite-1v8u.onrender.com/)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,7 +54,12 @@
 ### Key Features <a name="key-features"></a>
```
Comment (thecodechaser) [inline README.md:55]: - [ ] I was not able to run the project on my local, it's showing the below errors. Please make sure to fix the issues and users should be able to run the project on the local. Also make sure that all tests are passing.

image

Comment (thecodechaser) [inline README.md:55]: - [ ] On the 'Categories page' I'm not able to see the uploaded icons, It's showing broken images or nothing. Please make sure to fix this issue and users should be able to see the categories images. (If you're facing issues with uploading images, you can also provide option for users to add image link).

image

Comment (thecodechaser) [inline README.md:55]: - [ ] Please make sure that 'Add new category' and 'Add new transaction' buttons match with the below design as requested in the project requirements to follow the design guidelines.

### Your design

image

### Given design 

image

Comment (thecodechaser) [inline README.md:55]: - [ ] On the transaction page I can't find any option to add transactions(I'm not sure what the categories details page is I don't think it's a transaction page, I also tried to open the transaction page with routes but it didn't work). Please make sure to add a transaction page and on the transaction page to add an option so users are able to add transactions for that category. (After fixing the issue, make sure that everything is working as per project requirements, the next reviewer will check all the other features that I'm not able to check).

image

Comment (thecodechaser) [inline README.md:55]: - [ ] In the presentation video, Please make sure also to show some code that you want to highlight as requested in the project requirements. Currently, I can't see any highlighted code in the video.

image

Comment (thecodechaser) [inline README.md:55]: - [ ] In the Acknowledgments section of the README file, Please make sure that you give appropriate credit to the author [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) as requested in the project requirements.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @martinkarugaba👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Good job with validations. ✔️ 
- Nice UI design. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (franclobo) [reply README.md:55]: Dear @thecodechaser when you have this issue please 🙏🏽 run `rails assets:precompile` on your terminal. In Ruby on Rails, asset precompilation is a process that converts your application's assets (such as stylesheets and JavaScript files) into a format that can be served efficiently by a web server in a production environment. This process helps reduce the load time of your web pages by minimizing and optimizing your assets.

File: README.md [hunk 2]
```diff
@@ -146,12 +151,14 @@ Feel free to check the [issues page](../../issues/).
 
 ## 🙏 Acknowledgments <a name="acknowledgements"></a>
 
-I would like to thank [Microverse](https://www.microverse.org/?grsf=martin-9isn0t) for guiding me through this project, giving code reviews and a community of fellow coders.
+- [ ] I would like to thank [Microverse](https://www.microverse.org/?grsf=martin-9isn0t) for guiding me through this project, giving code reviews and a community of fellow coders.
+
+- [ ] Many thanks to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) for providing the [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=) for the app
```
Comment (franclobo) [inline README.md:156]: - [OPTIONAL] Something weird happened while I was reviewing although the acknowledgment to the author is added in the code I couldn't see it in the browser 👇🏽 Please 🙏🏽 ensure that it appears correctly in the browser.

image

File: README.md [hunk 3]
```diff
@@ -26,7 +26,7 @@
 
 **[budget_app]** is an app that helps you create transactions of different categories.
 
-[Link_to_video_description](https://www.loom.com/share/7dd6d32e1c2142ca884d8aa8b3a24372)
+[Link_to_video_description](https://www.loom.com/share/2caf14323b2b4d359b3c3c99a159a1c9?sid=d268d7d9-ca6d-4a80-9f8d-bfe2bf3d7a0c)
```
Comment (franclobo) [inline README.md:29]: - **WELL DONE!** 👏🏽 👏🏽 👏🏽

**Review by franclobo — APPROVED** [review]
Hi @martinkarugaba,

# STATUS: APPROVED 😃 

leonardo-dicaprio-leo-dicaprio

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

## Highlights

- No linter errors ✔️ 
- Good UI ✔️ 
- Professional documentation ✔️ 
- All tests passed ✔️ 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**